definitions/scenarios/self_upgrade.rb

```diff
@@ -65,13 +75,26 @@ class SelfUpgrade < SelfUpgradeBase
     end
 
     def downstream_self_upgrade(pkgs_to_update)
+      unless upgrade_allowed?
+        if version_difference > 1
+          raise(
+            ForemanMaintain::Error::Warn,
+            "foreman-maintain is too many versions ahead, given the" \
+            "target version is #{target_version}. Please rollback " \
+            "foreman-maintain to the proper version."
+          )
+        else
+          raise(
+            ForemanMaintain::Error::Warn,
+            "self-upgrade has already been executed, and is ready to upgrade to #{target_version}"
```

Comment:
We could opt to instead run `dnf update` to update to the latest foreman-maintain from the maintenance repository if for some reason the user self-upgraded and then a new maintain was released? That seems rather corner case but perhaps more in-line?

Reply:
Seems like a safe suggestion in case someone ran self upgrade but then left it alone for a while

Reply:
@evgeni do you concur?

Reply:
You mean something like `dnf --disablerepo=* --enablerepo=NEW_MAINT rubygem-foreman_maintain satellite-maintain`? Yeah… I could see this being useful.

Reply:
So, uh, actually exactly what we already do for "self upgrade" a few lines later.